examples/seismic/viscoacoustic/operators.py

```diff
@@ -80,40 +79,42 @@ def sls_1st_order(model, geometry, p, r=None, **kwargs):
     # Density
     rho = 1. / b
 
-    # Bulk modulus
-    bm = rho * vp**2
-
     # Attenuation Memory variable.
     r = r or TimeFunction(name="r", grid=model.grid, time_order=1,
                           space_order=space_order, save=save_t, staggered=NODE)
 
     if forward:
 
-        # Define PDE
-        pde_v = v - s * b * grad(p)
-        u_v = Eq(v.forward, damp * pde_v)
+        # Particle velocity
+        pde_v = v.dt + b * grad(p)
+        u_v = Eq(v.forward, damp * solve(pde_v, v.forward))
 
-        pde_r = r - s * (1. / t_s) * r - s * (1. / t_s) * tt * rho * div(v.forward)
-        u_r = Eq(r.forward, damp * pde_r)
+        # Attenuation Memory variable.
+        pde_r = r.dt + (1. / t_s) * (r + tt * rho * div(v.forward))
+        u_r = Eq(r.forward, damp * solve(pde_r, r.forward))
 
-        pde_p = p - s * bm * (tt + 1.) * div(v.forward) - s * vp**2 * r.forward + \
-            s * vp**2 * q
-        u_p = Eq(p.forward, damp * pde_p)
+        # Pressure
+        pde_p = m * p.dt + rho * (tt + 1.) * div(v.forward) + r.forward - q
+        u_p = Eq(p.forward, damp * solve(pde_p, p.forward))
 
         return [u_v, u_r, u_p]
 
     else:
 
-        # Define PDE
-        pde_r = r - s * (1. / t_s) * r - s * p
-        u_r = Eq(r.backward, damp * pde_r)
+        # Attenuation Memory variable.
+        pde_r = r.dt.T + (1. / t_s) * r + p
```

Comment:
No adjoint of `r.dt` is `r.dt.T` no minus sign

Reply:
So, the adjoint of `r.dt,` in discretized form, is `(r_{t - dt} - r_t)/dt` instead `(r_t - r_{t-dt})/dt` ?
I have always used the last one. But, following your logic, the `pde_r` shouldn't be 
` r.dt.T - (1. / t_s) * r - p` ? 
Please see the equation A-12 from this [paper](https://library.seg.org/doi/10.1190/geo2013-0414.1)

Take advantage of this discussion... should be equivalent `r.dtl `and `-r.dt.T`?
I could see that the `.evaluate` are the same, but the results are different when I tested these equations.

Reply:
If it's first order then you have `- r.dtl = r.dt.T` yes as used for the vector ones since the `.T` is the matriicial transpose.

Concerning `r` the PDe is correct, the PDE that correspond to `r.backward = r - s * (1. / t_s) * r - s * p` is `r.dt.T +  (1. / t_s) * r + p`.   I am only using the adjoint definitions we have defined (and tested) for derivative so that the adjoitn equation is what is expected.

Reply:
I see. But the issue is that when I wrote `r.backward = r - s * (1. / t_s) * r - s * p` I was considering that the adjoint of `r.dt` was `-r.dt.T`. 
Notice that if I had assumed that the adjoint of `r.dt` was `r.dt.T`, so the PDE should be `r.backward = r + s * (1. / t_s) * r + s * p` . 
Either way, both equations produce the same result. It's just a matter of writing.